stdlib/lexing.ml

```diff
@@ -163,10 +163,11 @@ let from_channel ?with_positions ic =
   from_function ?with_positions (fun buf n -> input ic buf 0 n)
 
 let from_string ?(with_positions = true) s =
+  (* [lex_buffer] is exported in the mli, one could mutate it outside this module. *)
```

Comment:
I find the comment a bit unclear: it explains why we need to make a copy, but without saying explicitly that we are (intentionally) making a copy.

Reply:
I've changed it a bit